### DISCUSSION TITLE
invisible TextLayer causes performance loss

### DISCUSSION THREAD

**Discussion by euzu** [visgl/deck.gl#7075] [Q&A] [answered]
I am using deck.gl 8.7.11, react-map-gl 6.1.19, react 18.2.0, maplibre-gl 2.1.9.

I have 3 layers, the first is a custom layer, the second is a scatterplot layer, the third is a TextLayer. I am displaying 100K items.  The TextLayer is only visible when zoomed in very close. 
Even if the text layer is invisible, it has a big impact on the performance. As soon as I remove the TextLayer completely, the performance improves. 

For all 3 layers I set ```dataComparator = () => false;``` because the data changes several times a second. I have a loop which sets the data every 250 ms. I don't have any updateTriggers.

I had read that layers should not be simply removed, but only made invisible. But in this case the performance impact is too big.

How can I solve this problem ?

**Comment by euzu** [comment, marked as answer]
i have found  [Data accessor methods shouldn't be invoked for hidden layers](https://github.com/visgl/deck.gl/issues/6056).
I have implemented the ```shouldUpdateState```  Method.
Now its seems to have no impact, but i dont know if there are now other issues, i couldnt see any.

**Reply by Pessimistress** [reply]
Invisible layers still update their attributes if `data` changes. I do not recommend using `dataComparator = () => false`. Just use a new `data` object every time the data does change.

**Reply by euzu** [reply]
thank you,  when invisible i will set an empty list as data.
